x-pack/metricbeat/module/sql/query/query.go

```diff
@@ -146,7 +145,13 @@ func dbSelector(driver, dbName string) string {
 	return ""
 }
 
-func (m *MetricSet) fetch(ctx context.Context, db *sql.DbClient, reporter mb.ReporterV2, queries []query) (bool, error) {
+// fetch returns a sanitized error recreated with errors.New (via defer sql.SanitizeError) to avoid leaking sensitive data.
+// The returned error does not wrap the original, so errors.Is/As will not match.
```

Comment:
This doesn't seem to hold true now, since we are including the err as well in return statement